packages/components/src/ClickableStyle/ClickableStyle.tsx

```diff
@@ -40,10 +67,43 @@ const ClickableStyle = React.forwardRef(
       state,
       variant,
       className,
+      icon,
       ...rest
     }: ClickableStyleProps<IComponent>,
     ref: React.ForwardedRef<HTMLElement>,
   ) => {
+    const maybeRenderIcon = (side: "left" | "right") => {
+      if (!icon) {
+        return;
+      }
+
+      const {
+        iconElement,
+        iconPlacement = "left",
+        iconHorizontalMargin = 0,
+      } = icon;
+
+      if (
+        (side === "left" && iconPlacement !== "left") ||
+        (side === "right" && iconPlacement !== "right")
```

Comment:
having the 2 different side props is confusing to me 🤔   might be cleaner to destructure `iconPlacement` beforehand and then do
```
iconPlacement === "left" && renderIcon(styles.iconLeft)
iconPlacement === "right" && renderIcon(styles.iconRight)
```